test/e2e/apps/job.go

```diff
@@ -818,57 +818,50 @@ var _ = SIGDescribe("Job", func() {
 		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
 
-		ginkgo.By("Ensuring job has ready pods")
-		err = e2ejob.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, ready)
+		ginkgo.By("Ensure the job controller updates the status.ready field")
+		err = e2ejob.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, ptr.To[int32](0))
 		framework.ExpectNoError(err, "failed to ensure job ready in namespace: %s", f.Namespace.Name)
 
-		ginkgo.By("Ensuring pods for job exist")
-		pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
-		framework.ExpectNoError(err, "failed to get pod list for job in namespace: %s", f.Namespace.Name)
-		podsReady := int32(0)
-		for _, pod := range pods.Items {
-			for _, c := range pod.Status.Conditions {
-				if c.Type == v1.PodReady && c.Status == v1.ConditionTrue {
-					podsReady++
-				}
+		ginkgo.By("Updating the job with suspend=false")
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			job, err = e2ejob.GetJob(ctx, f.ClientSet, f.Namespace.Name, job.Name)
+			framework.ExpectNoError(err, "unable to get job %s in namespace %s", job.Name, f.Namespace.Name)
+			job.Spec.Suspend = ptr.To[bool](false)
+			updatedJob, err := e2ejob.UpdateJob(ctx, f.ClientSet, f.Namespace.Name, job)
+			if err == nil {
+				job = updatedJob
 			}
-		}
-		gomega.Expect(podsReady).To(gomega.Equal(parallelism), "expected %d running job pods, but got %d", parallelism, podsReady)
-	})
-
-	ginkgo.It("should check the ready of a pods for failed job", func(ctx context.Context) {
-		backoff := 0
-		ginkgo.By("Creating a job")
-		job := e2ejob.NewTestJob("fail", "all-fail", v1.RestartPolicyNever, parallelism, completions, nil, int32(backoff))
-
-		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
-		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
+			return err
+		})
+		framework.ExpectNoError(err, "failed to update job in namespace: %s", f.Namespace.Name)
 
-		ginkgo.By("Ensuring job has ready pods")
-		err = e2ejob.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, 0)
+		ginkgo.By("Ensure the job controller updates the status.ready field")
+		err = e2ejob.WaitForJobReady(ctx, f.ClientSet, f.Namespace.Name, job.Name, &parallelism)
 		framework.ExpectNoError(err, "failed to ensure job ready in namespace: %s", f.Namespace.Name)
 
-		ginkgo.By("Ensuring pods for job exist")
-		pods, err := e2ejob.GetJobPods(ctx, f.ClientSet, f.Namespace.Name, job.Name)
-		framework.ExpectNoError(err, "failed to get pod list for job in namespace: %s", f.Namespace.Name)
-		podsReady := int32(0)
-		for _, pod := range pods.Items {
-			for _, c := range pod.Status.Conditions {
-				if c.Type == v1.PodReady && c.Status == v1.ConditionTrue {
-					podsReady++
-				}
+		ginkgo.By("Updating the job with suspend=true")
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Extract this to a utility function, maybe we call it `updateJobSuspend` or `updateJobSuspendWithRetries` to commonize the repetition in the test code

Reply:
added https://github.com/kubernetes/kubernetes/pull/121300/commits/a236969b197a353a58f6f6a244a43baa0efca60b